### PR TITLE
Update Astro to 0.26.1

### PR DESCRIPTION
**Pull request by IanVS**
This builds on https://github.com/DefinedNet/nebula-docs/pull/23 and updates astro to 0.26.1, as well as a small bump for netlify cms.  

The major changes here are outlined in the [migration guide](https://docs.astro.build/en/migrate/#migrate-to-v026), but here are the main things I needed to do:

- Changed `Astro.fetchContent()` to `Astro.glob()` and deal with the new markdown parser changes (frontmatter is collected into `frontmatter`, the `path` changed shape, etc.)
- `Astro.request.canonicalURL` changed to `Astro.canonicalURL`

I also hit a few bugs that I needed to work around:
- https://github.com/withastro/compiler/issues/378
- https://github.com/withastro/compiler/issues/379

Closes https://github.com/DefinedNet/nebula-docs/pull/23

### CODE REVIEW

File: src/components/ConfigOption.astro [hunk 1]
```diff
@@ -4,7 +4,9 @@ import { Code, Markdown } from 'astro/components';
...
-<slot/>
-{isRequired && <p class="mt-16 badge badge-info">Required</p>}
-{description?.trim() && <Markdown content={description}/>}
-{example && <Code code={example} lang='yaml' />}
+<div>
+  <slot/>
+  {isRequired && <p class="mt-16 badge badge-info">Required</p>}
+  {description?.trim() && <Markdown content={description}/>}
+  {example && <Code code={example} lang='yaml' />}
+</div>
```
Comment (IanVS) [inline src/components/ConfigOption.astro:12]: Wrapping in a div to workaround https://github.com/withastro/compiler/issues/379.

File: src/components/MobileNav.astro [hunk 2]
```diff
@@ -44,7 +44,7 @@ const currentPageTitle = pages.find(p => p.href === currentPage)?.title
 
 </style>
 
-<script type="module" hoist>
+<script type="module">
```
Comment (IanVS) [inline src/components/MobileNav.astro:47]: The cli told me we don't need `hoist` anymore.

File: tsconfig.json [hunk 3]
```diff
@@ -6,5 +6,7 @@
     "moduleResolution": "node",
     "resolveJsonModule": true,
     "allowSyntheticDefaultImports": true,
+    "isolatedModules": true,
+    "types": ["vite/client"],
```
Comment (IanVS) [inline tsconfig.json:10]: This aligns us with https://docs.astro.build/en/guides/typescript/#setup.

File: src/get-page-links.ts [hunk 4]
```diff
@@ -30,11 +30,11 @@ export function getPageLinks({currentPage, allPages}) {
 function buildPageObj(slug: string, langCode: string, allPages: any) {
   // Find the right file and the right language
   let page = allPages.find(p => 
-    getSlugFromFilename(p.file.pathname) === slug && getLanguageFromFilename(p.file.pathname) === langCode
+    getSlugFromFilename(p.file) === slug && getLanguageFromFilename(p.file) === langCode
   );
   // If there wasn't a translated version, fall back to default.
   if (!page) {
-    page = allPages.find(p => getSlugFromFilename(p.file.pathname) === slug && getLanguageFromFilename(p.file.pathname) === DEFAULT_LOCALE);
+    page = allPages.find(p => getSlugFromFilename(p.file) === slug && getLanguageFromFilename(p.file) === DEFAULT_LOCALE);
```
Comment (jasikpark) [inline src/get-page-links.ts:37]: does this mean that `p.file` is now a string instead of a `File()`?

**Review by jasikpark — APPROVED** [review]
lgtm; we don't have percy or other ci so I just clicked around the dev server

File: src/get-page-links.ts (hunk 4 above)
Comment (IanVS) [reply src/get-page-links.ts:37]: Yes, it's an absolute path now. https://docs.astro.build/en/guides/markdown-content/#importing-markdown

**Pull request merged**